### ISSUE TITLE
SohoTreeComponent - Children property is undefined when tree is multi select.

### ISSUE THREAD

**Issue by mdevale** [infor-design/enterprise-ng#377]
**Describe the bug**
When getting tree dataset, children property of tree node who has children's before assigning dataset to tree is undefined when tree is multi select.

**To Reproduce**
1. Do the changes as described in additional context
2. Run the project
3. Browse tree-source page in tree control
4. See console output

**Expected behavior**
If tree node has children then children property of that node should not be undefined.

**Version**
- ids-enterprise-ng: v5.0.0 

**Screenshots**
![image](https://user-images.githubusercontent.com/40569892/53416045-a2f2bb00-39f8-11e9-8b72-995492f2f6d3.png)

**Platform**
 - OS Version: Windows 10
 - Browser Name - chrome
 - Browser Version - Version 72.0.3626.109 (Official Build) (64-bit)

**Additional context**
1. Put below code in tree-source.demo.ts to make tree multi select.  
    `treeOptions: SohoTreeOptions = {    
    hideCheckboxes: true,
    selectable: 'multiple'
   };

  ngOnInit() {
    this.tree.options = this.treeOptions;
  }`

2. Define ngAfterViewInit in tree-source.demo.ts as below
   ` ngAfterViewInit() {        
    const data = this.demoData.getRootTreeNodes();
    console.log('Assigned Dataset: ' + JSON.stringify(data));    
    this.tree.dataset = data;
    console.log('Dataset from tree: '); 
    console.log(this.tree.dataset);    
  }`

3. Change ROOT_NODE in tree-data.demo.ts as below
    `private ROOT_NODE: SohoTreeNode = {
    'id': 'Root ' + this.id++,
    'open': true,
    'text': 'Node 1',
    'children': [
        {
          'id': 'Node 1' + '/1',
          'text': 'Leaf ' + this.id++
        },
        {
          'id': 'Node 1' + '/2',
          'text': 'Node ' + this.id++,
          'children': []
        }
      ]
    };`

**Comment by tmcconechy** [comment]
@mdevale Can you just attach a zip with the tree-service.demo.ts updated and ready to drop in? I cant seem to make the code you mentioned compile.

**Comment by tmcconechy** [comment]
Im not too sure whats happening here but i see the point. I do get this error...

```
ERROR in src/app/tree/tree-service.demo.ts(39,23): error TS2339: Property 'demoData' does not exist on type 'TreeServiceDemoComponent'.
```

But it somehow still runs. Also i was thinking this might be a EP issue so tried this example in EP
```
  var sampleData = [{
    'id': 'Root 2 ',
    'open': true,
    'text': 'Node 1',
    'children': [
        {
          'id': 'Node 1' + '/1',
          'text': 'Leaf 1'
        },
        {
          'id': 'Node 1' + '/2',
          'text': 'Node 2',
          'children': []
        }
      ]
    }];

  //Can refer to the data set here
  $('#json-tree').tree({dataset: sampleData});
  console.log($('#json-tree').data('tree').settings.dataset)
```

Notice i pull the setttings.dataset. I cant seem to get this example to run and test that in NG. Maybe that helps tho if you change the example to use `this.tree.options.dataset` if you want to pull the data that way?

**Comment by mdevale** [comment]
Yes issue with EP. After loading tree, children property is there in dataset after statement self.syncDataset(); but after self.initSelected() children property does not exist in dataset. Please see attached screenshot. 

Before self.initSelected():
![image](https://user-images.githubusercontent.com/40569892/53467006-2bfe0680-3a7a-11e9-91ea-131c6d1cab60.png)

After self.initSelected():
![image](https://user-images.githubusercontent.com/40569892/53467056-6667a380-3a7a-11e9-9205-2036f4e5c0e7.png)

Issue with self.syncNode(a); statement in setNodeStatus() method which is being called from initSelected() method. If we comment this line, it works fine but i am not sure is this correct resolution?
Children also not available in `this.tree.options.dataset`

**Comment by mdevale** [comment]
@tmcconechy you can pull the fork "mdevale/enterprise-ng" and run. Then browse tree-dynamic page and click on Expand All button. See console output. "Node Two" has children's but you can't see in tree dataset.

**Comment by tmcconechy** [comment]
I think the problem might be in syncNode (which is called view initSelected) as you said. We shouldnt comment out the line, should try and make sure syncNode will copy the children over as well. Maybe something here..

```
    // Children
    const ul = nodeJQ.next();
    if (ul[0] && ul[0].tagName.toLowerCase() === 'ul') {
      entry.children = [];

      const items = [].slice.call(ul.children('li').toArray());
      items.forEach((li) => {
        entry.children.push(self.syncNode(li.querySelector('a')));
      });
    }
```

Can you move this to EP please @clepore ?

**Comment by mdevale** [comment]
it is copying children but it is deleted in subsequent code to code snippet mentioned by you in above comment.

`      // Merge json data
      var jsonData = nodeJQ.data('jsonData');
      if (jsonData) {
        delete jsonData.selected;
        delete jsonData.children;
        entry = $.extend({}, jsonData, entry);
      }`

**Comment by tmcconechy** [comment]
Ok makes sense - then maybe we just move the copy to after that action so it gets copied on. Did you want to make a PR?

**Comment by mdevale** [comment]
This is also not resolving the issue.
